crates/sui-types/src/gas.rs

```diff
@@ -467,7 +467,13 @@ pub fn deduct_gas(gas_object: &mut Object, deduct_amount: u64, rebate_amount: u6
     assert!(balance >= deduct_amount);
     let new_gas_coin = GasCoin::new(*gas_coin.id(), balance + rebate_amount - deduct_amount);
     let move_object = gas_object.data.try_as_move_mut().unwrap();
-    move_object.update_contents_and_increment_version(bcs::to_bytes(&new_gas_coin).unwrap());
+    // unwrap safe because GasCoin is guaranteed to serialize
+    let new_contents = bcs::to_bytes(&new_gas_coin).unwrap();
+    //debug_assert_eq!(move_object.type_specific_contents().len(), new_contents.len());
```

Comment:
Why is this commented out?

Reply:
> Seems like there are debug asserts in each of the callsites. Is it possible to dedup the code by moving them to the same function call?

Ended up getting rid of this and similar asserts because they are redundant--the right check happens in the callee.